src/components/PostDetails/PostDetails.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect } from 'react';
+import { NewCommentForm } from '../NewCommentForm';
+import './PostDetails.scss';
+import { Post, Comment } from '../../types';
+import { getComments, delComment } from '../../api/comments';
+import { getPostbyId } from '../../api/posts';
+
+type Props = {
+  postId: number | undefined;
+};
+
+export const PostDetails: React.FC<Props> = ({
+                                               postId,
+                                             }) => {
+  const [commentsList, setCommentsList] = useState<Comment[] | undefined>([]);
+  const [visiblecomments, setVisiblecomments] = useState(false);
+  const [postDetails, setPostDetails] = useState<Post>();
+
+  const findPost = async () => {
+    if (postId) {
+      const result = await getPostbyId(postId);
+
+      setPostDetails(result);
+    }
+  };
+
+  useEffect(() => {
+    findPost();
+  }, []);
+
+  const findcomments = async () => {
+    if (postId) {
+      const result = await getComments(postId);
+
+      setCommentsList(result);
+    }
+  };
+
+  const deletecomment = async (id: number) => {
+    if (commentsList) {
+      await delComment(id);
+      findcomments();
+    }
+  };
+
+  useEffect(() => {
+    findcomments();
+  }, [postId]);
+
+  return (
+    <div className="PostDetails">
+      <h2>Post details:</h2>
+      {postId !== 0 && (
+        <>
+          <section className="PostDetails__post">
+            <p>
+              {postDetails?.title}
+            </p>
+          </section>
+          <section className="PostDetails__comments">
+            <button
+              type="button"
+              className="button"
+              onClick={() => {
+                setVisiblecomments(!visiblecomments);
+              }}
+            >
+              {commentsList && ((visiblecomments)
```

Comment:
The logic is inverted. When the comments are visible, the button displays "show". 

```suggestion
              {commentsList && (visiblecomments
  ? `Hide ${commentsList.length} comments`
  : `Show ${commentsList.length} comments`)}

```